test/Transformations_test.jl

```diff
@@ -188,57 +188,36 @@
     end
 
     @testset "ColumnReduction" begin
-        using Tenet: ColumnReduction, find_zero_columns
+        using Tenet: ColumnReduction
 
-        @testset "rank reduction" begin
+        @testset "range" begin
             data = rand(3, 3, 3)
-            data[:, 1:2, :] .= 0 # 1st and 2nd column of the 2nd dimension are zero
-            # Since there is only one non-zero column, the whole 2nd dimension can be reduced
+            data[:, 1:2, :] .= 0
 
             A = Tensor(data, (:i, :j, :k))
             B = Tensor(rand(3, 3), (:j, :l))
             C = Tensor(rand(3, 3), (:j, :m))
 
-            @test issetequal(find_zero_columns(parent(A)), [(2, 1), (2, 2)])
-
             tn = TensorNetwork([A, B, C])
             reduced = transform(tn, ColumnReduction)
 
-            # Test that all the tensors in reduced have no columns and they do not have the 2nd :j index
-            for tensor in tensors(reduced)
-                @test isempty(find_zero_columns(parent(tensor)))
-                @test :j ∉ inds(tensor)
-            end
-
-            @test length(tn.indices) > length(reduced.indices)
-
-            # Test that the resulting contraction is the same as the original
-            @test contract(reduced) ≈ contract(contract(A, B; dims = Symbol[]), C)
+            @test :j ∉ inds(reduced)
+            @test contract(reduced) ≈ contract(tn)
         end
 
-        @testset "index size reduction" begin
+        @testset "int" begin
             data = rand(3, 3, 3)
-            data[:, 2, :] .= 0 # 2nd column of the 2nd dimension can be reduced
+            data[:, 2, :] .= 0
 
             A = Tensor(data, (:i, :j, :k))
             B = Tensor(rand(3, 3), (:j, :l))
             C = Tensor(rand(3, 3), (:j, :m))
 
-            @test issetequal(find_zero_columns(parent(A)), [(2, 2)])
-
             tn = TensorNetwork([A, B, C])
             reduced = transform(tn, ColumnReduction)
 
-            # Test that all the tensors in reduced have no columns and they have smaller dimensions in the 2nd :j index
-            for tensor in tensors(reduced)
-                @test isempty(Tenet.find_zero_columns(parent(tensor)))
-                @test size(tensor, :j) == 2
-            end
-
-            @test length(tn.indices) == length(reduced.indices)
-
-            # Test that the resulting contraction is the same as the original
-            @test contract(reduced) ≈ view(contract(tn), :j => 1:2:3)
+            @test size(reduced, :j) == 2
+            @test contract(reduced) ≈ contract(tn)
```

Comment:
Okay, be careful here since if the `:j` index is not contracted, this line would not work and we would need something like `contract(reduced) ≈ view(contract(tn), :j => 1:2:3)`.

Reply:
I don't see why. We designed the case and we are setting to contract because it appears more than once in `tn`.